diesel/src/query_builder/select_statement/boxed.rs

```diff
@@ -95,6 +95,17 @@ impl<'a, ST, QS, DB> QueryFragment<DB> for BoxedSelectStatement<'a, ST, QS, DB>
         try!(self.offset.collect_binds(out));
         Ok(())
     }
+
+    fn is_safe_to_cache_prepared(&self) -> bool {
+        self.distinct.is_safe_to_cache_prepared() &&
+            self.select.is_safe_to_cache_prepared() &&
+            self.from.from_clause().is_safe_to_cache_prepared() &&
+            self.where_clause.as_ref().map(|w| w.is_safe_to_cache_prepared())
```

Comment:
Ah I thought we were in a different module. Yeah, `where_clause` is absolutely at a different level of abstraction in this case. It's treated specially because it's the only case where the equivalent query builder method is append and not replace (e.g. `filter(foo).filter(bar)` is the same as `filter(foo.and(bar))`, while `order(foo).order(bar)` is equivalent to `order(bar)`).

The fact that we're boxed makes things a little tricky. Actually making it uniform would require doing something like `Box<QueryFragment<DB> + WhereAnd<Box<QueryFragment<DB> + WhereAnd<Box<QueryFragment<DB> ...`, and infinitely recursive types like that are illegal. I'm not super happy with it, but don't have a great answer. It's a problem with this module as a whole though, not just this specific method.